### PR TITLE
Add flannel v0.22.2 template

### PR DESCRIPTION
**Pull request by rayandas**
Related: https://github.com/rancher/rancher/issues/41840

### CODE REVIEW

File: pkg/rke/templates/flannel_v0.22.2.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  verbs:
+  - list
+  - watch
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  labels:
+	k8s-app: flannel
+  name: flannel
```
Comment (jiaqiluo) [inline pkg/rke/templates/flannel_v0.22.2.go]: This needs to be fixed at multiple places:
```suggestion
metadata:
  labels:
    k8s-app: flannel
  name: flannel
```

File: pkg/rke/templates/flannel_v0.22.2.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
+package templates
+
+/*
+FlannelTemplateV0_22_2 is based on upstream flannel v0.22.2
+Source: https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml
+Upstream Changelog:
+- added get verb for nodes in flannel ClusterRole
+- added clustercidrs resources list and watch permission for flannel ClusterRole.
```
Comment (jiaqiluo) [inline pkg/rke/templates/flannel_v0.22.2.go:8]:
```suggestion
- added clustercidrs resources list and watch permission for flannel ClusterRole.
- added k8s-app label
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
I have a few comments. The rest looks good.

File: pkg/rke/templates/flannel_v0.22.2.go (hunk 1 above)
Comment (rayandas) [reply pkg/rke/templates/flannel_v0.22.2.go]: Updated the PR with suggested changes

File: pkg/rke/templates/flannel_v0.22.2.go (hunk 2 above)
Comment (rayandas) [reply pkg/rke/templates/flannel_v0.22.2.go:8]: Updated the PR with suggested changes

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**